src/pages/academy/xpCalculation/XpCalculation.tsx

```diff
@@ -0,0 +1,249 @@
+  };
+
+  const updatePaginationState = () => {
+    if (gridApi) {
+      const newTotalPages = gridApi.paginationGetTotalPages();
+      const newCurrPage = newTotalPages === 0 ? 0 : gridApi.paginationGetCurrentPage() + 1;
+      setPageState({
+        currPage: newCurrPage,
+        maxPages: newTotalPages,
+        rowCountString: formatRowCountString(
+          25,
+          newCurrPage,
+          newTotalPages,
+          gridApi.paginationGetRowCount()
+        ),
+        isBackDisabled: newTotalPages === 0 || newCurrPage === 1,
+        isForwardDisabled: newTotalPages === 0 || newCurrPage === newTotalPages
+      });
+    }
+  };
+
+  // const handleFilterChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+  //   const changeVal = event.target.value;
+  //   setFilterText(changeVal);
+  // };
+
+  // const handleApplyFilter = () => {
+  //   if (gridApi) {
+  //     gridApi.setQuickFilter(filterText);
+  //   }
+  // };
+
+  // const handleFilterKeypress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+  //   if (event.key === 'Enter') {
+  //     handleApplyFilter();
+  //   }
+  // };
+
+  const onGridReady = (params: GridReadyEvent) => {
+    setGridApi(params.api);
+    updatePaginationState();
+  };
+
+  const Controls = () => {
+    return (
+      <div className="grading-controls">
+        <div className="GridControls ag-grid-controls">
+          <div style={{ width: '130px' }} className="left-controls"></div>
+          <div className="centre-controls">
+            <Button
+              icon={IconNames.CHEVRON_BACKWARD}
+              onClick={changePaginationView('first')}
+              minimal={true}
+              disabled={pageState.isBackDisabled}
+            />
+            <Button
+              icon={IconNames.CHEVRON_LEFT}
+              onClick={changePaginationView('prev')}
+              minimal={true}
+              disabled={pageState.isBackDisabled}
+            />
+            <Button className="pagination-details hidden-xs" disabled={true} minimal={true}>
+              <div>{`Page ${pageState.currPage} of ${pageState.maxPages}`}</div>
+              <div>{pageState.rowCountString}</div>
+            </Button>
+            <Button
+              icon={IconNames.CHEVRON_RIGHT}
+              onClick={changePaginationView('next')}
+              minimal={true}
+              disabled={pageState.isForwardDisabled}
+            />
+            <Button
+              icon={IconNames.CHEVRON_FORWARD}
+              onClick={changePaginationView('last')}
+              minimal={true}
+              disabled={pageState.isForwardDisabled}
+            />
+          </div>
+          <div className="right-controls">
+            <Button icon={IconNames.EXPORT} onClick={exportCSV}>
+              <span className="hidden-xs">Export to CSV</span>
+            </Button>
+          </div>
+        </div>
+      </div>
+    );
+  }; /* Display either a loading screen or a table with overviews. */
```

Comment:
Would it be better if this component is moved out of `XpCalculation`? This component looks very similar to the ones used in `src/pages/academy/grading/Grading.tsx`. I think it may be reasonable to abstract this further to reduce the code duplication.

Reply:
let's KIV for the future and get this merged in first to prepare for the sem